tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,318 @@
+    fi
+
+    local retry=0
+    while : ; do
+        echo "waiting for all ${appname} pods are re-created"
+        local actual_count=$(kubectl -n "${NAMESPACE}" get pod -l app="${appname}" "${status_option}" --no-headers 2>/dev/null | wc -l)
+
+        if [ "${desired_count}" = "${actual_count}" ]; then
+           break
+        fi
+
+        if [ "${retry}" = "${RETRY_COUNT}" ] ; then
+          echo 'timed out'
+            exit 1
+        fi
+        sleep "${RETRY_PERIOD}"
+        retry=$((retry+1))
+    done
+}
+
+function restore_manifests() {
+    BACKUP_MON_DEPLOYMENTS=$(ls "${BACKUP_DIR}"/rook-ceph-mon-*.yaml 2>/dev/null)
+    DESIRED_MON_COUNT=$(echo "${BACKUP_MON_DEPLOYMENTS}" | wc -w)
+
+    for mon in ${BACKUP_MON_DEPLOYMENTS}; do
+        kubectl replace --force -f "${mon}"
+    done
+    for i in ${MON_IDS}; do
+        kubectl delete -n "${NAMESPACE}" pod -l app=rook-ceph-mon -l mon="${i}"
+    done
+    wait_pods rook-ceph-mon "${DESIRED_MON_COUNT}" Running
+
+    BACKUP_OSD_DEPLOYMENTS=$(ls "${BACKUP_DIR}"/rook-ceph-osd-*.yaml 2>/dev/null)
+    DESIRED_OSD_COUNT=$(echo "${BACKUP_OSD_DEPLOYMENTS}" | wc -w)
+
+    for osd in ${BACKUP_OSD_DEPLOYMENTS}; do
+        kubectl replace --force -f "${osd}"
+    done
+    for i in ${OSD_IDS}; do
+        kubectl delete -n "${NAMESPACE}" pod -l app=rook-ceph-osd -l osd="${i}"
+    done
+
+    BACKUP_TOOL_DEPLOYMENT=$(ls "${BACKUP_DIR}"/rook-ceph-tools.yaml 2>/dev/null)
+    kubectl replace --force -f "${BACKUP_TOOL_DEPLOYMENT}"
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-tools
+    wait_pods rook-ceph-tools 1 Running
+    TOOL=$(kubectl get -n "${NAMESPACE}" pod -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+}
+
+function restore() {
+    set +e
+    for i in ${MON_IDS}; do
+        echo "Restoring monitor store of mon ${i}"
+        name=$(kubectl get -n "${NAMESPACE}" pod -l app=rook-ceph-mon -l mon="${i}" -o json | jq -r '.items[].metadata.name')
+        STORE_DB=/var/lib/ceph/mon/ceph-"${i}"/store.db
+        kubectl exec -i -n "${NAMESPACE}" "${name}" -- bash -c "test -d ${STORE_DB}.corrupted && rm -rf ${STORE_DB} && mv ${STORE_DB}.corrupted ${STORE_DB}"
+    done
+    restore_manifests
+
+    kubectl -n "${NAMESPACE}" scale deployment rook-ceph-operator --replicas=1
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-operator
+    wait_pods rook-ceph-operator 1 Running
+
+    DESIRED_MGR_COUNT=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-mgr --no-headers 2>/dev/null | wc -l)
+    kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-mgr
+    wait_pods rook-ceph-mgr "$DESIRED_MGR_COUNT" Running
+}
+
+function init() {
+    local exist_kubectl=1
+    which kubectl || exist_kubectl=0
+    if [ "${exist_kubectl}" = 0 ]; then
+        echo "'kubectl' must exist" >&2
+        exit 1
+    fi
+    local connectable=1
+    kubectl get pod >/dev/null 2>&1 || connectable=0
+    if [ "${connectable}" = 0 ]; then
```

Comment:
ditto use `if ! command` instead

Reply:
This `if` statement is not for checking command. I'll use `if ! command` for checking `jq`'s existence.

Reply:
You can still do `if kubectl get pod` instead of checking the value of `connectable`.

Reply:
Indeed, will fix.